public/cron.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+// Semaphone part is meant to prevent concurrent execution
+// I dropped the idea of using real semaphores because a lot of 
+// providers don't allow it.
+// TODO : Maybe create a semaphore class with a fallback on file locks
+// TODO : Ensure no locks remain there for long (would block cron)
+// TODO : make a javascript more clever doing periodic calls
+//        + push here a json structure with information so that javascript call does not come back too often
+//        if there are a lot of people connected
+
+// Get the semaphore
+$fp = fopen("cronlock.txt", "w+");
+
+// If semaphore can not be created, exit
+if (!flock($fp, LOCK_EX)) {
+    exit;
+}
+
+define('RESTRICTED', FALSE);
+define('ROOT', dirname(__FILE__));
+
+include_once '../config/configuration.php';
+include_once '../config/appSettings.php';
+include_once '../src/core/class.autoload.php';
+
+use leantime\domain\repositories;
+
+$config = new leantime\core\config();
+$settings = new leantime\core\appSettings();
+$settings->loadSettings($config->defaultTimezone);
+
+// TODO  check if using the session class in cron is a better idea
+session_start();
+
+if(isset($config->appUrl) && $config->appUrl != ""){
+    define('BASE_URL', $config->appUrl);
+    define('CURRENT_URL', $config->appUrl.$settings->getRequestURI($config->appUrl));
+} else{
+    define('BASE_URL', $settings->getBaseURL());
+    define('CURRENT_URL', $settings->getFullURL());
+}
+
+ob_start();
+
+if ( $_GET['mode'] == "debug" )
+{
+    $DEBUG_CRON=true;
+} else
+{
+    $DEBUG_CRON=false;
+}
+
+function debug_print ($message)
+{
+    global $DEBUG_CRON;
+    if ( $DEBUG_CRON == true )
+    {
+        print($message."\n");
+    }
+}
+
+// Fake template to be replaced by something better
+// TODO : Rework email templating system
+function doFormatMail ($messageToSendToUser)
+{
+    $outputHTML="Leantime has news for you...<br/>\n";
```

Comment:
If you use the standard mail function from class.mailer.php you don't need to format anything. It is already preformatted

Reply:
Well ... for the global formatting of the mail yes... but I was thinking of the content... maybe people will want the various notices in the mail to be presented in a different way...